### ISSUE TITLE
Select all (data library) no longer restricts to current page

### ISSUE THREAD

**Issue by gregvonkuster** [galaxyproject/galaxy#13917]
Describe the bug & reproduce

Go in a library folder that contains more than the default / manually specified page size datasets
Use the select all check box to select datasets
All datasets in the entire folder are selected

Expected behavior
Only datasets that are displayed in the current page should be selected

I believe this is a recent change, and I'm not sure if it was intended.  We used to be able to specify the folder page size and click the Select All checkbox to select all datasets in the current page.  But now the Select All checkbox selects all datasets in the entire folder.  This is a problem with folders containing many datasets, but only a portion are wanted (e.g., it's difficult to check 300 boxes separately if you only want to retrieve 300 of 5000).

If this is intended, then perhaps we could have an additional option to select only the current page.

**Comment by wm75** [comment]
I guess it's semi-intended ;)
For *very* large data libraries trying to show all records so that you can import them all would take a *very* long time otherwise.
I agree that if there was a intuitive, clear way of saying "select all currently displayed" vs "select all in folder" that would be optimal.

**Comment by gregvonkuster** [comment]
> I agree that if there was a intuitive, clear way of saying "select all currently displayed" vs "select all in folder" that would be optimal.

Agreed!   We need the ability to go both ways (select all / select displayed) for large folders.  :)

**Comment by dannon** [comment]
The new (current dev) history multi select and bulk operations feel very
intuitive to me; we can try to generalize that and apply them to data
libraries too.

**Comment by ahmedhamidawan** [comment]
> Only datasets that are displayed in the current page should be selected

With recent modernizations, this is how it currently works. Now, I wonder if we still need an option to select _everything_ in a library.
